### PR TITLE
added config reload [POC]

### PR DESCRIPTION
**Pull request by XK4MiLX**
- implemented config reload base on fs.watch
- improved ui api call
-  fix typo in dockerContainerExec
- fix bad look button

### CODE REVIEW

File: HomeUI/src/views/fluxadmin/ManageFlux.vue [hunk 1]
```diff
@@ -308,7 +308,7 @@
             </b-button>
             <confirm-dialog
               target="update-blockedRepositories"
-              confirm-button="Update Blocked Repositories"
```
Comment (Cabecinha84) [inline HomeUI/src/views/fluxadmin/ManageFlux.vue:311]: It was correct like it was before, its blocked repositories. Please change it.

File: ZelBack/config/default.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
-const userconfig = require('../../config/userconfig');
+let userconfig = require('../../config/userconfig'); // eslint-disable-line
```
Comment (Cabecinha84) [inline ZelBack/config/default.js:1]: what is the  // eslint-disable-line for? if not needed please remove it.

File: ZelBack/config/default.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
 
-const isDevelopment = userconfig.initial.development || false;
+const isDevelopment = userconfig.initial.development || false; // eslint-disable-line
```
Comment (Cabecinha84) [inline ZelBack/config/default.js:4]: what is the  // eslint-disable-line for? if not needed please remove it.

File: apiServer.js [hunk 4]
```diff
@@ -10,13 +11,47 @@ const app = require('./ZelBack/src/lib/server');
...
+        if (md5Current === md5Previous) {
+          return;
+        }
+        md5Previous = md5Current
+        if( initialMd5 === null) {
+          initialMd5 = md5Current
+        }
+        if ( initialMd5 !== md5Current ) {
+          initialMd5 = null
+          console.log(`Config file changed, reloading ${event.filename}...`);
```
Comment (Cabecinha84) [inline apiServer.js:40]: no console.log, change to log.info()

File: apiServer.js [hunk 5]
```diff
@@ -10,13 +11,47 @@ const app = require('./ZelBack/src/lib/server');
...
+      }
+    }
+  } catch (error) {
+      console.error(`Error watching files: ${error}`);
+  }
+}
+
+setInterval(async () => {
+  configReload();
+}, 2 * 1000);
```
Comment (Cabecinha84) [inline apiServer.js:53]: lets make it slower, 60 * 1000
once per minute is more than enough.

File: app.js [hunk 6]
```diff
@@ -12,15 +13,49 @@ const app = require('./ZelBack/src/lib/server');
...
+        if (md5Current === md5Previous) {
+          return;
+        }
+        md5Previous = md5Current
+        if( initialMd5 === null) {
+          initialMd5 = md5Current
+        }
+        if ( initialMd5 !== md5Current ) {
+          initialMd5 = null
+          console.log(`Config file changed, reloading ${event.filename}...`);
```
Comment (Cabecinha84) [inline app.js:44]: log.info instead

File: app.js [hunk 7]
```diff
@@ -12,15 +13,49 @@ const app = require('./ZelBack/src/lib/server');
...
+    }
+  } catch (error) {
+      console.error(`Error watching files: ${error}`);
+  }
+}
+
+setInterval(async () => {
+  configReload();
+}, 2 * 1000);
+
```
Comment (Cabecinha84) [inline app.js:58]: once every 60 seconds instead

File: apiServer.js [hunk 8]
```diff
@@ -10,13 +11,47 @@ const app = require('./ZelBack/src/lib/server');
...
+        if ( initialMd5 !== md5Current ) {
+          initialMd5 = null
+          console.log(`Config file changed, reloading ${event.filename}...`);
+        }
+        delete require.cache[require.resolve('./config/userconfig')];
+        userconfig = require('./config/userconfig');
+      }
+    }
+  } catch (error) {
+      console.error(`Error watching files: ${error}`);
```
Comment (Cabecinha84) [inline apiServer.js:47]: use log.error() instead.

File: app.js [hunk 9]
```diff
@@ -12,15 +13,49 @@ const app = require('./ZelBack/src/lib/server');
...
+        if ( initialMd5 !== md5Current ) {
+          initialMd5 = null
+          console.log(`Config file changed, reloading ${event.filename}...`);
+        }
+        delete require.cache[require.resolve('./config/userconfig')];
+        userconfig = require('./config/userconfig');
+      }
+    }
+  } catch (error) {
+      console.error(`Error watching files: ${error}`);
```
Comment (Cabecinha84) [inline app.js:51]: use log.error() instead.

File: package.json [hunk 10]
```diff
@@ -92,6 +92,7 @@
     "http-shutdown":"~1.2.2",
     "inquirer": "~8.2.5",
     "lru-cache": "~10.0.1",
+    "md5": "~2.3.0",
```
Comment (Cabecinha84) [inline package.json:95]: I don't think it's needed to add the md5 lib. Check if hash lib works as we already have it included.
example to search for on the code: "const messageHash = hash(msg);"

**Review by Cabecinha84 — CHANGES_REQUESTED** [review]
Good job. Requesting some changes.

File: HomeUI/src/views/fluxadmin/ManageFlux.vue (hunk 1 above)
Comment (XK4MiLX) [reply HomeUI/src/views/fluxadmin/ManageFlux.vue:311]: check other buttons it is too long.. look bad ( big and without space between buttons )
image

File: ZelBack/config/default.js (hunk 2 above)
Comment (XK4MiLX) [reply ZelBack/config/default.js:1]: without it you won't compile home ui and it will throw an error

File: ZelBack/config/default.js (hunk 3 above)
Comment (XK4MiLX) [reply ZelBack/config/default.js:4]: without it you won't compile home ui and it will throw an error

File: apiServer.js (hunk 5 above)
Comment (XK4MiLX) [reply apiServer.js:53]: i dont agree it should work fast ( hot reload ) same way as working nodemon

File: package.json (hunk 10 above)
Comment (XK4MiLX) [reply package.json:95]: good idea :) tested working correct

File: apiServer.js (hunk 4 above)
Comment (XK4MiLX) [reply apiServer.js]: replaced :)

File: app.js (hunk 6 above)
Comment (XK4MiLX) [reply app.js]: replaced :)

File: apiServer.js (hunk 8 above)
Comment (XK4MiLX) [reply apiServer.js]: replaced :)

File: apiServer.js (hunk 5 above)
Comment (XK4MiLX) [reply apiServer.js:52]: or set it global as 30s  and import function and use it when any cmd with config is use

File: HomeUI/src/views/fluxadmin/ManageFlux.vue (hunk 1 above)
Comment (Cabecinha84) [reply HomeUI/src/views/fluxadmin/ManageFlux.vue:311]: agree makes sense

File: apiServer.js (hunk 5 above)
Comment (Cabecinha84) [reply apiServer.js:52]: ok let it be every 2 seconds

**Review by Cabecinha84 — APPROVED** [review]
Ack

**Pull request merged**